File: sample-chat-obj-c/sample-chat/Components/ChatScreen/Views/ProgressView/ProgressView.m
//
//  ProgressView.m
//  sample-chat
//
//  Created by Injoit on 17.02.2022.
//  Copyright © 2022 Quickblox. All rights reserved.
//

#import "ProgressView.h"
#import "UIView+Chat.h"

@interface ProgressView()
@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *activityIndicator;
@end

@implementation ProgressView

- (void)awakeFromNib {
    [super awakeFromNib];
    
    [self setRoundViewWithCornerRadius:16.0f];
}

- (BOOL)isRunning {
    return self.activityIndicator.isAnimating;
}

- (void)start {
    if ([self isRunning]) {
        return;
    }
    UIWindow *lastWindow = UIApplication.sharedApplication.windows.lastObject;
    [lastWindow addSubview:self];
    self.center = lastWindow.center;
    dispatch_async(dispatch_get_main_queue(), ^{
        [self.activityIndicator startAnimating];
    });
}

- (void)stop {
    dispatch_async(dispatch_get_main_queue(), ^{
        [self.activityIndicator stopAnimating];
        [self removeFromSuperview];
    });
}

@end

File: sample-chat-swift/sample-chat-swift/Components/Utils/ProgressView/ProgressView.swift
//
//  ProgressView.swift
//  sample-chat-swift
//
//  Created by Injoit on 09.11.2021.
//  Copyright © 2021 QuickBlox. All rights reserved.
//

import UIKit

class ProgressView: UIView {
    
    @IBOutlet private weak var activityIndicator: UIActivityIndicatorView!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        rounded(cornerRadius: 16)
        layer.zPosition = CGFloat(Float.greatestFiniteMagnitude)
    }
    
    var isRunning: Bool {
        return activityIndicator.isAnimating
    }
    
    func start() {
        if isRunning == true { return }
        guard let lastWindow = UIApplication.shared.windows.last else { return }
        lastWindow.addSubview(self)
        center = lastWindow.center
        lastWindow.bringSubviewToFront(self)
        DispatchQueue.main.async {
            self.activityIndicator.startAnimating()
        }
    }
    
    func stop() {
        DispatchQueue.main.async {
            self.activityIndicator.stopAnimating()
            self.removeFromSuperview()
        }
    }
}
